exporter/datadogexporter/translate_traces.go

```diff
@@ -232,6 +232,11 @@ func spanToDatadogSpan(s pdata.Span,
 		}
 	}
 
+	//  canonical resource attribute version should override others if it exists
+	if rsTagVersion := tags[conventions.AttributeServiceVersion]; rsTagVersion != "" {
+		tags[versionTag] = rsTagVersion
+	}
+
```

Comment:
Nit: this code block (along with the lines above) could be refactored later. Right now it reads as:
- take the version from the config
- but if there’s a version tag, use it
- but if there’s a resource attribute version, use it instead

which is a bit counterintuitive. Doing checks in priority order (resource attribute, then version tag, then config) could make this clearer.

Reply:
👍  agreed, cleaned this up a bit, should be a bit more readable now